clone_roles.py

```diff
@@ -62,11 +62,21 @@ def get_issue_by_title(title: str, issues: List[Issue]):
     def _rotate_roles(self) -> List[str]:
         maintainers = [issue.assignees[0].login for issue in self.previous_week_issues]
         candidates = list(PEOPLE - set(maintainers))
-
-        # remove responsible for the first role
-        maintainers.pop(0)
-        # randomly pick someone who didn't do anything last week ;)
-        maintainers.append(random.choice(candidates))
+        random.shuffle(candidates)
+
+        # Remove the responsible of the first role and use the last
+        # of the candidates for the last one.
+        maintainers = maintainers[1:] + [candidates.pop()]
+        # Some maintainers who were active might not
+        # be in the PEOPLE set already.
+        # Replace them with someone from candidates.
+        # Chose someone randomly if candidates run out.
+        for i, maintainer in enumerate(maintainers):
+            if maintainer not in PEOPLE:
+                try:
+                    maintainers[i] = candidates.pop()
+                except IndexError:
+                    maintainers[i] = random.choice(list(PEOPLE))
```

Comment:
```suggestion
                maintainers[i] = candidates.pop() if candidates else random.choice(list(PEOPLE))
```

Reply:
I couldn't resist and factored this out as `next_candidate()`, and use it instead of the previous `candidates.pop()`, too.